roiextractors/extractors/nwbextractors/nwbextractors.py

```diff
@@ -164,50 +254,223 @@ def get_num_channels(self):
+
+                # Write to file
+                io.write(nwbfile)
+        else:
+            NwbImagingExtractor.add_devices(imaging=imaging,
+                                            nwbfile=nwbfile,
+                                            metadata=metadata)
+
+            NwbImagingExtractor.add_two_photon_series(imaging=imaging,
+                                                      nwbfile=nwbfile,
+                                                      metadata=metadata)
+
+            NwbImagingExtractor.add_epochs(imaging=imaging,
+                                           nwbfile=nwbfile)
 
 
 class NwbSegmentationExtractor(SegmentationExtractor):
-    """
-    Class used to extract data from the NWB data format. Also implements a
-    static method to write any format specific object to NWB.
-    """
 
-    def __init__(self, filepath, optical_channel_name=None,
-                 imaging_plane_name=None, image_series_name=None,
-                 processing_module_name=None,
-                 neuron_roi_response_series_name=None,
-                 background_roi_response_series_name=None):
+    extractor_name = 'NwbSegmentationExtractor'
+    installed = True  # check at class level if installed or not
+    is_writable = False
+    mode = 'file'
+    installation_mesg = ""  # error message when not installed
+
+    def __init__(self, file_path):
         """
+        Creating NwbSegmentationExtractor object from nwb file
         Parameters
         ----------
-        filepath: str
-            The location of the folder containing dataset.nwb file.
-        optical_channel_name: str(optional)
-            optical channel to extract data from
-        imaging_plane_name: str(optional)
-            imaging plane to extract data from
-        image_series_name: str(optional)
-            imaging series to extract data from
-        processing_module_name: str(optional)
-            processing module to extract data from
-        neuron_roi_response_series_name: str(optional)
-            name of roi response series to extract data from
-        background_roi_response_series_name: str(optional)
-            name of background roi response series to extract data from
+        file_path: str
+            .nwb file location
         """
         check_nwb_install()
-        if not os.path.exists(filepath):
+        SegmentationExtractor.__init__(self)
+        if not os.path.exists(file_path):
             raise Exception('file does not exist')
 
-        self.filepath = filepath
+        self.file_path = file_path
         self.image_masks = None
-        self.pixel_masks = None
         self._roi_locs = None
         self._accepted_list = None
-        io = NWBHDF5IO(filepath, mode='r+')
-        nwbfile = io.read()
+        self._io = NWBHDF5IO(file_path, mode='r+')
+        nwbfile = self._io.read()
         self.nwbfile = nwbfile
         _nwbchildren_type = [type(i).__name__ for i in nwbfile.all_children()]
```

Comment:
Since we know the structure of the NWB file, I'd rather access the different fields that we know should be there. E.g. `nwbfile['ophys']['OpticalChannel']` and so on. It would improve readibility

Reply:
thanks for this suggestion, I have made another implementation and created a small PR altogether in #46